File: 75_SortColors.c
#include<stdio.h>

void sortColors(int* nums, int numsSize)
{
    int count[3] = {0};
    int output[numsSize];
    
    for(int i = 0; i < numsSize; i++)
    {
        count[nums[i]]++;
    }
    for(int i = 1; i < 3; i++)
    {
        count[i] += count[i - 1];
    }
    for(int i = 0; i < numsSize; i++)
    {
        output[--count[nums[i]]] = nums[i];
    }
    for(int i = 0; i < numsSize; i++)
    {
        nums[i] = output[i];
    }
}
int main()
{
    int arr[] = {2,0,2,1,1,0};
    sortColors(arr,6);

    return 0;
}

File: 75_SortColors.js


var sortColors = function(nums) {
    
    let min_index = 0,i = 0,j = 0;

    for(i = 0; i < nums.length-1; i++)
    {
        min_index = i;
        for(j = i+1; j < nums.length; j++)
        {
            if(nums[j] < nums[min_index])
            {
                min_index = j;
            }
        }
        let temp = nums[i];
        nums[i] = nums[min_index];
        nums[min_index] = temp;
    }
    return nums;
};

let main = function()
{
    let arr = [2,0,2,1,1,0];
    console.log(sortColors(arr));
}

main();